src/main/java/jsr223/kubernetes/utils/BindingUtils.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package jsr223.kubernetes.utils;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.script.ScriptContext;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+
+@NoArgsConstructor
+public class BindingUtils {
+
+    // GI, bindings and variables
+    @Getter
+    Map<String, String> environment;
+
+    private void addMapBindingAsEnvironmentVariable(String bindingKey, Map<?, ?> bindingValue,
```

Comment:
please rename these methods, for example to addMapBindingToReplacementMap(String bindingKey, Map<?, ?> mapBindingValue, Map<String, String> replacementMap)

etc...

Reply:
The thing is, not only replacements var will be extracted from the bindings - there is also others GI used to customize the behavior.  That's why I kept originally the 'environment' name.

Reply:
Eventually renamed to addMapBindingToEngineMetadata, etc.

Reply:
ok why not